sos/report/plugins/krb5.py

```diff
@@ -33,10 +35,32 @@ def setup(self):
             f"{self.kdcdir}/kdc.conf",
             "/var/log/kadmind.log"
         ])
+        self.collect_kinit()
         self.add_copy_spec("/var/log/krb5kdc.log", tags="kerberos_kdc_log")
         self.add_cmd_output(f"klist -ket {self.kdcdir}/.k5*")
         self.add_cmd_output("klist -ket /etc/krb5.keytab")
 
+    def collect_kinit(self):
+        """
+        Collect the kinit command output for the system with id_provider "AD"
+        or "IPA" domains.
+        """
+
+        hostname = socket.gethostname()[:11].upper()
```

Comment:
Why the last 11 characters? I'm not super familiar with kerberos myself, but it seems odd to take a statically sized slice here to me.

Reply:
@Be-Rahul below is a simple thing that would cause problems of just using 11 characters, and hence you would not get the full hostname

```
ubuntu@juju-728f4e-13-lxd-3:~$ python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> socket.gethostname()
'juju-728f4e-13-lxd-3'
>>> socket.gethostname()[:11]
'juju-728f4e'
>>> socket.gethostname()[:11].upper()
'JUJU-728F4E'
```

It would be better to split on the `.` if it is FQDN , and if there isn't, just get the full hostname

Reply:
Hello @arif-ali , I had discussion with IDM SME and there is limit of 15 characters in the command used for AD.
I used the above suggestion as.

for id_provider = ipa
hostname = socket.getfqdn()

for id_provider = ad
hostname = hostname.split('.')[0][:15].upper()